docs/design/storage-auto-scale.md

```diff
@@ -102,6 +120,10 @@ Controller:
    
     1) Verify the Storagecluster whether the new osds are added or scaled in size, for all the device sets.
     
+       1) For vertical scaling, Query osd size from Prometheus and match it with storagecluster.spec..size.
```

Comment:
Will we use the same prometheus query from the goroutine to get the osd size and count? Hopefully we can at least share some code there to make sure we are getting consistent values.

Reply:
And if the values change for osd size/count, the syncmap needs to be updated here.

Reply:
Yes are already querying the prometheus query to get osd size and count, we can use the same code with go-routine.

> And if the values change for osd size/count, the syncmap needs to be updated here.

Should we just give the scrapper the responsibility to update the sync map (every 10 mins),
 As if we do it here, all individual crs will update the syncmap in parallel.

Reply:
I would assume we read from the cache for this?